src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1662,29 +1662,58 @@ class Simplex_tree {
  private:
   bool rec_prune_above_filtration(Siblings* sib, Filtration_value filt) {
     auto&& list = sib->members();
-    auto last = std::remove_if(list.begin(), list.end(), [this,filt](Dit_value_t& simplex) {
+    bool modified;
+
+    if constexpr (Options::stable_simplex_handles) {
+      modified = false;
+      for (auto sh = list.begin(); sh != list.end();) {
+        if (sh->second.filtration() > filt) {
+          modified = true;
+          if (has_children(sh)) rec_delete(sh->second.children());
+          // dimension may need to be lowered
+          dimension_to_be_lowered_ = true;
+          sh = list.erase(sh);
+        } else {
+          ++sh;
+        }
+      }
+      if (list.empty() && sib != root()) {
+        // Removing the whole siblings, parent becomes a leaf.
+        sib->oncles()->members()[sib->parent()].assign_children(sib->oncles());
+        delete sib;
+        // dimension may need to be lowered
+        dimension_to_be_lowered_ = true;  //TODO: already done for sure, no?
+        return true;
+      } else {
+        // Keeping some elements of siblings. Remove the others, and recurse in the remaining ones.
+        for (auto&& simplex : list)
+          if (has_children(&simplex)) modified |= rec_prune_above_filtration(simplex.second.children(), filt);
+      }
+    } else {
+      auto last = std::remove_if(list.begin(), list.end(), [this, filt](Dit_value_t& simplex) {
         if (simplex.second.filtration() <= filt) return false;
         if (has_children(&simplex)) rec_delete(simplex.second.children());
         // dimension may need to be lowered
         dimension_to_be_lowered_ = true;
         return true;
       });
 
-    bool modified = (last != list.end());
-    if (last == list.begin() && sib != root()) {
-      // Removing the whole siblings, parent becomes a leaf.
-      sib->oncles()->members()[sib->parent()].assign_children(sib->oncles());
-      delete sib;
-      // dimension may need to be lowered
-      dimension_to_be_lowered_ = true;
-      return true;
-    } else {
-      // Keeping some elements of siblings. Remove the others, and recurse in the remaining ones.
-      list.erase(last, list.end());
-      for (auto&& simplex : list)
-        if (has_children(&simplex))
-          modified |= rec_prune_above_filtration(simplex.second.children(), filt);
+      modified = (last != list.end());
+      if (last == list.begin() && sib != root()) {
+        // Removing the whole siblings, parent becomes a leaf.
+        sib->oncles()->members()[sib->parent()].assign_children(sib->oncles());
+        delete sib;
+        // dimension may need to be lowered
+        dimension_to_be_lowered_ = true;
+        return true;
+      } else {
+        // Keeping some elements of siblings. Remove the others, and recurse in the remaining ones.
+        list.erase(last, list.end());
```

Comment:
I did a benchmark, using random rips filtrations, on the current version and a factorized version (with the `erase` outside) like you suggested. The results are below. The times includes the construction of the Rips complex, as repeating the pruning without rebuilding the tree makes no sense.

<details><summary>Benchmark details (v1)</summary>

```C++
  bool rec_prune_above_filtration(Siblings* sib, Filtration_value filt) {
    auto&& list = sib->members();
    bool modified = false;

    if constexpr (Options::stable_simplex_handles) {
      for (auto sh = list.begin(); sh != list.end();) {
        if (sh->second.filtration() <= filt) {
          ++sh;
        } else {
          modified = true;
          if (has_children(sh)) rec_delete(sh->second.children());
          // dimension may need to be lowered
          dimension_to_be_lowered_ = true;
          sh = list.erase(sh);
        }
      }
    } else {
      auto last = std::remove_if(list.begin(), list.end(), [this, filt](Dit_value_t& simplex) {
        if (simplex.second.filtration() <= filt) return false;
        if (has_children(&simplex)) rec_delete(simplex.second.children());
        // dimension may need to be lowered
        dimension_to_be_lowered_ = true;
        return true;
      });

      modified = (last != list.end());
      list.erase(last, list.end());
    }

    if (list.empty() && sib != root()) {
      // Removing the whole siblings, parent becomes a leaf.
      sib->oncles()->members()[sib->parent()].assign_children(sib->oncles());
      delete sib;
      // dimension may need to be lowered
      dimension_to_be_lowered_ = true;  // TODO: already done for sure, or can the list be empty from the start?
      return true;
    } else {
      // Keeping some elements of siblings. Remove the others, and recurse in the remaining ones.
      for (auto&& simplex : list)
        if (has_children(&simplex)) modified |= rec_prune_above_filtration(simplex.second.children(), filt);
    }

    return modified;
  }
```

| nber_of_points/max_dim/threshold*10 | Time ((new - old) / \|old\|) | CPU       | Time Old (µs)                 | Time New (µs) | CPU Old (µs) | CPU New (µs) |
|-------------------------------------|------------------------------|-----------|-------------------------------|---------------|--------------|--------------|
| Prune_non_stable/100/6/2_pvalue     | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/6/2_mean       | +0.1308                      | +0.1302   | 43                            | 49            | 43           | 49           |
| Prune_non_stable/100/6/2_median     | +0.0667                      | +0.0660   | 43                            | 46            | 43           | 46           |
| Prune_non_stable/100/6/2_stddev     | +13.1198                     | +13.2071  | 0                             | 7             | 0            | 7            |
| Prune_non_stable/100/6/2_cv         | +11.4861                     | +11.5702  | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/6/2_pvalue     | 0.0140                       | 0.0140    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/6/2_mean       | +0.0697                      | +0.0694   | 175                           | 187           | 174          | 186          |
| Prune_non_stable/256/6/2_median     | +0.0746                      | +0.0743   | 163                           | 175           | 163          | 175          |
| Prune_non_stable/256/6/2_stddev     | -0.0520                      | -0.0520   | 25                            | 24            | 25           | 24           |
| Prune_non_stable/256/6/2_cv         | -0.1137                      | -0.1136   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/6/2_pvalue    | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/6/2_mean      | +0.0247                      | +0.0244   | 2013                          | 2063          | 2009         | 2058         |
| Prune_non_stable/1024/6/2_median    | +0.0431                      | +0.0427   | 1881                          | 1962          | 1877         | 1957         |
| Prune_non_stable/1024/6/2_stddev    | -0.0148                      | -0.0145   | 329                           | 325           | 328          | 323          |
| Prune_non_stable/1024/6/2_cv        | -0.0386                      | -0.0379   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/6/2_pvalue    | 0.0257                       | 0.0257    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/6/2_mean      | -0.0269                      | -0.0270   | 7440                          | 7240          | 7423         | 7223         |
| Prune_non_stable/2000/6/2_median    | +0.0435                      | +0.0433   | 6945                          | 7247          | 6931         | 7231         |
| Prune_non_stable/2000/6/2_stddev    | -0.9614                      | -0.9618   | 1048                          | 40            | 1043         | 40           |
| Prune_non_stable/2000/6/2_cv        | -0.9603                      | -0.9607   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/8/2_pvalue     | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/8/2_mean       | +0.0745                      | +0.0744   | 46                            | 49            | 46           | 49           |
| Prune_non_stable/100/8/2_median     | +0.0812                      | +0.0811   | 43                            | 46            | 43           | 46           |
| Prune_non_stable/100/8/2_stddev     | -0.3105                      | -0.3105   | 10                            | 7             | 10           | 7            |
| Prune_non_stable/100/8/2_cv         | -0.3584                      | -0.3583   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/8/2_pvalue     | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/8/2_mean       | +0.1384                      | +0.1380   | 162                           | 185           | 162          | 184          |
| Prune_non_stable/256/8/2_median     | +0.0662                      | +0.0661   | 163                           | 173           | 162          | 173          |
| Prune_non_stable/256/8/2_stddev     | +26.6926                     | +27.0489  | 1                             | 24            | 1            | 23           |
| Prune_non_stable/256/8/2_cv         | +23.3260                     | +23.6467  | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/8/2_pvalue    | 0.0140                       | 0.0140    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/8/2_mean      | -0.0369                      | -0.0369   | 2165                          | 2085          | 2160         | 2080         |
| Prune_non_stable/1024/8/2_median    | -0.0363                      | -0.0364   | 2036                          | 1962          | 2032         | 1958         |
| Prune_non_stable/1024/8/2_stddev    | -0.0780                      | -0.0779   | 296                           | 273           | 295          | 272          |
| Prune_non_stable/1024/8/2_cv        | -0.0426                      | -0.0425   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/8/2_pvalue    | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/8/2_mean      | +0.0155                      | +0.0154   | 7603                          | 7721          | 7587         | 7704         |
| Prune_non_stable/2000/8/2_median    | +0.0211                      | +0.0211   | 7111                          | 7262          | 7097         | 7247         |
| Prune_non_stable/2000/8/2_stddev    | -0.1014                      | -0.1013   | 1120                          | 1007          | 1115         | 1002         |
| Prune_non_stable/2000/8/2_cv        | -0.1150                      | -0.1149   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/10/2_pvalue    | 0.0022                       | 0.0022    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/10/2_mean      | +0.0582                      | +0.0582   | 47                            | 50            | 47           | 50           |
| Prune_non_stable/100/10/2_median    | +0.0608                      | +0.0608   | 44                            | 47            | 44           | 47           |
| Prune_non_stable/100/10/2_stddev    | -0.0154                      | -0.0153   | 9                             | 9             | 9            | 9            |
| Prune_non_stable/100/10/2_cv        | -0.0695                      | -0.0694   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/10/2_pvalue    | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/10/2_mean      | -0.0128                      | -0.0128   | 179                           | 177           | 179          | 177          |
| Prune_non_stable/256/10/2_median    | +0.0429                      | +0.0427   | 168                           | 175           | 167          | 175          |
| Prune_non_stable/256/10/2_stddev    | -0.8071                      | -0.8072   | 37                            | 7             | 37           | 7            |
| Prune_non_stable/256/10/2_cv        | -0.8046                      | -0.8047   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/10/2_pvalue   | 0.0022                       | 0.0022    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/10/2_mean     | -0.0404                      | -0.0405   | 2149                          | 2063          | 2145         | 2058         |
| Prune_non_stable/1024/10/2_median   | -0.0164                      | -0.0166   | 2006                          | 1973          | 2002         | 1969         |
| Prune_non_stable/1024/10/2_stddev   | -0.3541                      | -0.3542   | 449                           | 290           | 447          | 289          |
| Prune_non_stable/1024/10/2_cv       | -0.3269                      | -0.3269   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/10/2_pvalue   | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/10/2_mean     | +0.0925                      | +0.0922   | 7081                          | 7736          | 7067         | 7718         |
| Prune_non_stable/2000/10/2_median   | +0.0255                      | +0.0253   | 7079                          | 7260          | 7066         | 7244         |
| Prune_non_stable/2000/10/2_stddev   | +68.6988                     | +69.7845  | 22                            | 1509          | 21           | 1502         |
| Prune_non_stable/2000/10/2_cv       | +62.7962                     | +63.8099  | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/6/4_pvalue     | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/6/4_mean       | +0.0166                      | +0.0166   | 53                            | 54            | 53           | 54           |
| Prune_non_stable/100/6/4_median     | +0.0467                      | +0.0467   | 49                            | 51            | 49           | 51           |
| Prune_non_stable/100/6/4_stddev     | -0.0870                      | -0.0872   | 11                            | 10            | 11           | 10           |
| Prune_non_stable/100/6/4_cv         | -0.1019                      | -0.1021   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/6/4_pvalue     | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/6/4_mean       | -0.0079                      | -0.0080   | 216                           | 215           | 216          | 214          |
| Prune_non_stable/256/6/4_median     | +0.0611                      | +0.0608   | 202                           | 215           | 202          | 214          |
| Prune_non_stable/256/6/4_stddev     | -0.9936                      | -0.9936   | 45                            | 0             | 45           | 0            |
| Prune_non_stable/256/6/4_cv         | -0.9936                      | -0.9936   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/6/4_pvalue    | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/6/4_mean      | +0.0177                      | +0.0178   | 2777                          | 2826          | 2770         | 2819         |
| Prune_non_stable/1024/6/4_median    | +0.0163                      | +0.0165   | 2605                          | 2648          | 2599         | 2642         |
| Prune_non_stable/1024/6/4_stddev    | +0.1840                      | +0.1839   | 480                           | 568           | 477          | 565          |
| Prune_non_stable/1024/6/4_cv        | +0.1633                      | +0.1632   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/6/4_pvalue    | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/6/4_mean      | +0.0069                      | +0.0069   | 15053                         | 15157         | 15015        | 15119        |
| Prune_non_stable/2000/6/4_median    | +0.0090                      | +0.0090   | 14083                         | 14209         | 14049        | 14176        |
| Prune_non_stable/2000/6/4_stddev    | -0.3465                      | -0.3463   | 3073                          | 2009          | 3058         | 1999         |
| Prune_non_stable/2000/6/4_cv        | -0.3509                      | -0.3508   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/8/4_pvalue     | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/8/4_mean       | +0.1126                      | +0.1123   | 47                            | 52            | 47           | 52           |
| Prune_non_stable/100/8/4_median     | +0.0465                      | +0.0464   | 47                            | 49            | 47           | 49           |
| Prune_non_stable/100/8/4_stddev     | +27.4689                     | +27.7361  | 0                             | 10            | 0            | 10           |
| Prune_non_stable/100/8/4_cv         | +24.5879                     | +24.8338  | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/8/4_pvalue     | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/8/4_mean       | +0.0325                      | +0.0324   | 222                           | 230           | 222          | 229          |
| Prune_non_stable/256/8/4_median     | +0.0361                      | +0.0360   | 208                           | 216           | 208          | 215          |
| Prune_non_stable/256/8/4_stddev     | -0.2544                      | -0.2541   | 45                            | 34            | 45           | 34           |
| Prune_non_stable/256/8/4_cv         | -0.2778                      | -0.2775   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/8/4_pvalue    | 0.0257                       | 0.0257    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/8/4_mean      | -0.0434                      | -0.0432   | 2971                          | 2842          | 2964         | 2836         |
| Prune_non_stable/1024/8/4_median    | +0.0185                      | +0.0186   | 2791                          | 2843          | 2785         | 2836         |
| Prune_non_stable/1024/8/4_stddev    | -0.9929                      | -0.9931   | 495                           | 4             | 493          | 3            |
| Prune_non_stable/1024/8/4_cv        | -0.9926                      | -0.9928   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/8/4_pvalue    | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/8/4_mean      | +0.0155                      | +0.0155   | 14735                         | 14962         | 14696        | 14924        |
| Prune_non_stable/2000/8/4_median    | +0.0182                      | +0.0182   | 13806                         | 14057         | 13772        | 14023        |
| Prune_non_stable/2000/8/4_stddev    | -0.0901                      | -0.0900   | 2775                          | 2525          | 2762         | 2513         |
| Prune_non_stable/2000/8/4_cv        | -0.1039                      | -0.1039   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/10/4_pvalue    | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/10/4_mean      | +0.0436                      | +0.0433   | 54                            | 56            | 54           | 56           |
| Prune_non_stable/100/10/4_median    | +0.0498                      | +0.0495   | 50                            | 53            | 50           | 53           |
| Prune_non_stable/100/10/4_stddev    | -0.2556                      | -0.2557   | 11                            | 8             | 11           | 8            |
| Prune_non_stable/100/10/4_cv        | -0.2867                      | -0.2866   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/10/4_pvalue    | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/10/4_mean      | +0.0304                      | +0.0304   | 225                           | 232           | 225          | 232          |
| Prune_non_stable/256/10/4_median    | +0.0316                      | +0.0316   | 211                           | 218           | 211          | 218          |
| Prune_non_stable/256/10/4_stddev    | -0.1584                      | -0.1584   | 45                            | 38            | 45           | 38           |
| Prune_non_stable/256/10/4_cv        | -0.1832                      | -0.1832   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/10/4_pvalue   | 0.0140                       | 0.0140    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/10/4_mean     | +0.0502                      | +0.0502   | 2838                          | 2980          | 2831         | 2973         |
| Prune_non_stable/1024/10/4_median   | +0.0040                      | +0.0041   | 2787                          | 2798          | 2780         | 2792         |
| Prune_non_stable/1024/10/4_stddev   | +2.4588                      | +2.4594   | 169                           | 583           | 168          | 581          |
| Prune_non_stable/1024/10/4_cv       | +2.2936                      | +2.2942   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/10/4_pvalue   | 0.0022                       | 0.0022    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/10/4_mean     | +0.0180                      | +0.0181   | 13981                         | 14233         | 13945        | 14198        |
| Prune_non_stable/2000/10/4_median   | +0.0108                      | +0.0109   | 13219                         | 13361         | 13186        | 13330        |
| Prune_non_stable/2000/10/4_stddev   | +0.1237                      | +0.1237   | 2472                          | 2778          | 2460         | 2765         |
| Prune_non_stable/2000/10/4_cv       | +0.1038                      | +0.1037   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/6/6_pvalue     | 0.0257                       | 0.0257    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/6/6_mean       | -0.0254                      | -0.0255   | 62                            | 60            | 62           | 60           |
| Prune_non_stable/100/6/6_median     | +0.0426                      | +0.0422   | 58                            | 60            | 58           | 60           |
| Prune_non_stable/100/6/6_stddev     | -0.9904                      | -0.9906   | 9                             | 0             | 9            | 0            |
| Prune_non_stable/100/6/6_cv         | -0.9902                      | -0.9903   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/6/6_pvalue     | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/6/6_mean       | +0.0256                      | +0.0255   | 282                           | 289           | 282          | 289          |
| Prune_non_stable/256/6/6_median     | +0.0269                      | +0.0269   | 265                           | 272           | 264          | 271          |
| Prune_non_stable/256/6/6_stddev     | +0.2734                      | +0.2737   | 44                            | 56            | 44           | 56           |
| Prune_non_stable/256/6/6_cv         | +0.2417                      | +0.2420   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/6/6_pvalue    | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/6/6_mean      | +0.0939                      | +0.0937   | 6260                          | 6847          | 6245         | 6830         |
| Prune_non_stable/1024/6/6_median    | +0.0264                      | +0.0264   | 6262                          | 6427          | 6247         | 6412         |
| Prune_non_stable/1024/6/6_stddev    | +136.2632                    | +136.5994 | 10                            | 1364          | 10           | 1357         |
| Prune_non_stable/1024/6/6_cv        | +124.4861                    | +124.8126 | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/6/6_pvalue    | 0.3075                       | 0.3075    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/6/6_mean      | +0.0098                      | +0.0099   | 165535                        | 167163        | 165195       | 166832       |
| Prune_non_stable/2000/6/6_median    | +0.0048                      | +0.0050   | 166473                        | 167277        | 166122       | 166949       |
| Prune_non_stable/2000/6/6_stddev    | -0.9204                      | -0.9200   | 8573                          | 682           | 8556         | 685          |
| Prune_non_stable/2000/6/6_cv        | -0.9212                      | -0.9208   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/8/6_pvalue     | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/8/6_mean       | +0.0386                      | +0.0385   | 64                            | 67            | 64           | 67           |
| Prune_non_stable/100/8/6_median     | +0.0417                      | +0.0416   | 60                            | 63            | 60           | 63           |
| Prune_non_stable/100/8/6_stddev     | +0.3827                      | +0.3837   | 9                             | 13            | 9            | 13           |
| Prune_non_stable/100/8/6_cv         | +0.3313                      | +0.3323   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/8/6_pvalue     | 0.0140                       | 0.0140    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/8/6_mean       | +0.0208                      | +0.0208   | 287                           | 293           | 286          | 292          |
| Prune_non_stable/256/8/6_median     | +0.0231                      | +0.0230   | 269                           | 275           | 268          | 274          |
| Prune_non_stable/256/8/6_stddev     | +0.0144                      | +0.0151   | 40                            | 40            | 39           | 40           |
| Prune_non_stable/256/8/6_cv         | -0.0063                      | -0.0056   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/8/6_pvalue    | 0.0036                       | 0.0036    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/8/6_mean      | +0.0070                      | +0.0071   | 8598                          | 8658          | 8577         | 8638         |
| Prune_non_stable/1024/8/6_median    | +0.0086                      | +0.0086   | 8584                          | 8658          | 8564         | 8637         |
| Prune_non_stable/1024/8/6_stddev    | -0.5172                      | -0.5164   | 61                            | 29            | 60           | 29           |
| Prune_non_stable/1024/8/6_cv        | -0.5206                      | -0.5198   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/8/6_pvalue    | 0.8501                       | 0.8501    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/8/6_mean      | +0.0185                      | +0.0185   | 224828                        | 228992        | 224342       | 228496       |
| Prune_non_stable/2000/8/6_median    | -0.0002                      | -0.0001   | 219089                        | 219042        | 218645       | 218627       |
| Prune_non_stable/2000/8/6_stddev    | +1.0353                      | +1.0358   | 26755                         | 54456         | 26629        | 54211        |
| Prune_non_stable/2000/8/6_cv        | +0.9983                      | +0.9988   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/10/6_pvalue    | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/10/6_mean      | +0.0384                      | +0.0383   | 62                            | 64            | 62           | 64           |
| Prune_non_stable/100/10/6_median    | +0.0414                      | +0.0414   | 58                            | 61            | 58           | 60           |
| Prune_non_stable/100/10/6_stddev    | -0.2247                      | -0.2246   | 11                            | 8             | 11           | 8            |
| Prune_non_stable/100/10/6_cv        | -0.2534                      | -0.2533   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/10/6_pvalue    | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/10/6_mean      | +0.0221                      | +0.0220   | 279                           | 285           | 278          | 285          |
| Prune_non_stable/256/10/6_median    | +0.0219                      | +0.0219   | 262                           | 267           | 261          | 267          |
| Prune_non_stable/256/10/6_stddev    | +0.1676                      | +0.1673   | 37                            | 43            | 37           | 43           |
| Prune_non_stable/256/10/6_cv        | +0.1423                      | +0.1422   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/10/6_pvalue   | 0.0211                       | 0.0211    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/10/6_mean     | -0.0249                      | -0.0243   | 8515                          | 8303          | 8489         | 8283         |
| Prune_non_stable/1024/10/6_median   | +0.0096                      | +0.0103   | 7967                          | 8044          | 7942         | 8024         |
| Prune_non_stable/1024/10/6_stddev   | -0.4641                      | -0.4646   | 1538                          | 824           | 1532         | 821          |
| Prune_non_stable/1024/10/6_cv       | -0.4504                      | -0.4513   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/10/6_pvalue   | 0.5205                       | 0.5205    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/10/6_mean     | +0.0230                      | +0.0231   | 356429                        | 364631        | 355630       | 363834       |
| Prune_non_stable/2000/10/6_median   | +0.0153                      | +0.0155   | 338431                        | 343612        | 337704       | 342937       |
| Prune_non_stable/2000/10/6_stddev   | +0.5579                      | +0.5584   | 46341                         | 72194         | 46092        | 71830        |
| Prune_non_stable/2000/10/6_cv       | +0.5228                      | +0.5232   | 0                             | 0             | 0            | 0            |
| OVERALL_GEOMEAN                     | +0.0280                      | +0.0279   | 0                             | 0             | 0            | 0            |

</details> 

<details><summary>Benchmark details (v2, with boolean to avoid calling erase everytime)</summary>

```C++
  bool rec_prune_above_filtration(Siblings* sib, Filtration_value filt) {
    auto&& list = sib->members();
    bool modified = false;
    bool emptied = false;

    if constexpr (Options::stable_simplex_handles) {
      for (auto sh = list.begin(); sh != list.end();) {
        if (sh->second.filtration() <= filt) {
          ++sh;
        } else {
          modified = true;
          if (has_children(sh)) rec_delete(sh->second.children());
          // dimension may need to be lowered
          dimension_to_be_lowered_ = true;
          sh = list.erase(sh);
        }
      }
      emptied = list.empty() && sib != root();
    } else {
      auto last = std::remove_if(list.begin(), list.end(), [this, filt](Dit_value_t& simplex) {
        if (simplex.second.filtration() <= filt) return false;
        if (has_children(&simplex)) rec_delete(simplex.second.children());
        // dimension may need to be lowered
        dimension_to_be_lowered_ = true;
        return true;
      });

      modified = (last != list.end());
      if (last != list.begin() || sib == root())  list.erase(last, list.end());
      else emptied = true;
    }

    if (emptied) {
      // Removing the whole siblings, parent becomes a leaf.
      sib->oncles()->members()[sib->parent()].assign_children(sib->oncles());
      delete sib;
      // dimension may need to be lowered
      dimension_to_be_lowered_ = true;  // TODO: already done for sure, or can the list be empty from the start?
      return true;
    } else {
      // Keeping some elements of siblings. Remove the others, and recurse in the remaining ones.
      for (auto&& simplex : list)
        if (has_children(&simplex)) modified |= rec_prune_above_filtration(simplex.second.children(), filt);
    }

    return modified;
  }
```

| nber_of_points/max_dim/threshold*10 | Time ((new - old) / \|old\|) | CPU       | Time Old (µs)                 | Time New (µs) | CPU Old (µs) | CPU New (µs) |
|-------------------------------------|------------------------------|-----------|-------------------------------|---------------|--------------|--------------|
| Prune_non_stable/100/6/2_pvalue     | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/6/2_mean       | +0.1574                      | +0.1566   | 43                            | 50            | 43           | 50           |
| Prune_non_stable/100/6/2_median     | +0.0861                      | +0.0854   | 43                            | 47            | 43           | 47           |
| Prune_non_stable/100/6/2_stddev     | +42.8424                     | +42.7169  | 0                             | 7             | 0            | 7            |
| Prune_non_stable/100/6/2_cv         | +36.8789                     | +36.7981  | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/6/2_pvalue     | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/6/2_mean       | +0.0754                      | +0.0751   | 174                           | 187           | 174          | 187          |
| Prune_non_stable/256/6/2_median     | +0.0786                      | +0.0783   | 163                           | 176           | 163          | 176          |
| Prune_non_stable/256/6/2_stddev     | -0.0891                      | -0.0890   | 27                            | 25            | 27           | 24           |
| Prune_non_stable/256/6/2_cv         | -0.1530                      | -0.1526   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/6/2_pvalue    | 0.0022                       | 0.0022    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/6/2_mean      | +0.0835                      | +0.0834   | 1936                          | 2097          | 1931         | 2092         |
| Prune_non_stable/1024/6/2_median    | +0.0700                      | +0.0699   | 1847                          | 1976          | 1842         | 1971         |
| Prune_non_stable/1024/6/2_stddev    | +0.1255                      | +0.1257   | 284                           | 319           | 282          | 318          |
| Prune_non_stable/1024/6/2_cv        | +0.0387                      | +0.0391   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/6/2_pvalue    | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/6/2_mean      | +0.0551                      | +0.0551   | 6817                          | 7193          | 6803         | 7178         |
| Prune_non_stable/2000/6/2_median    | +0.0559                      | +0.0560   | 6812                          | 7192          | 6797         | 7178         |
| Prune_non_stable/2000/6/2_stddev    | -0.4212                      | -0.4126   | 31                            | 18            | 30           | 18           |
| Prune_non_stable/2000/6/2_cv        | -0.4514                      | -0.4433   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/8/2_pvalue     | 0.0022                       | 0.0022    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/8/2_mean       | +0.0993                      | +0.0990   | 46                            | 50            | 46           | 50           |
| Prune_non_stable/100/8/2_median     | +0.0695                      | +0.0693   | 43                            | 46            | 43           | 46           |
| Prune_non_stable/100/8/2_stddev     | +0.3631                      | +0.3632   | 9                             | 13            | 9            | 13           |
| Prune_non_stable/100/8/2_cv         | +0.2400                      | +0.2404   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/8/2_pvalue     | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/8/2_mean       | +0.0513                      | +0.0512   | 174                           | 183           | 174          | 183          |
| Prune_non_stable/256/8/2_median     | +0.0546                      | +0.0545   | 163                           | 172           | 163          | 172          |
| Prune_non_stable/256/8/2_stddev     | -0.1699                      | -0.1698   | 36                            | 30            | 36           | 30           |
| Prune_non_stable/256/8/2_cv         | -0.2104                      | -0.2103   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/8/2_pvalue    | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/8/2_mean      | +0.0477                      | +0.0474   | 2021                          | 2118          | 2017         | 2112         |
| Prune_non_stable/1024/8/2_median    | +0.0536                      | +0.0533   | 1892                          | 1993          | 1888         | 1989         |
| Prune_non_stable/1024/8/2_stddev    | -0.3396                      | -0.3390   | 412                           | 272           | 410          | 271          |
| Prune_non_stable/1024/8/2_cv        | -0.3696                      | -0.3689   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/8/2_pvalue    | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/8/2_mean      | -0.0130                      | -0.0129   | 7493                          | 7395          | 7476         | 7379         |
| Prune_non_stable/2000/8/2_median    | +0.0387                      | +0.0387   | 7013                          | 7285          | 6999         | 7270         |
| Prune_non_stable/2000/8/2_stddev    | -0.7750                      | -0.7750   | 1551                          | 349           | 1544         | 347          |
| Prune_non_stable/2000/8/2_cv        | -0.7721                      | -0.7721   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/10/2_pvalue    | 0.0022                       | 0.0022    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/10/2_mean      | +0.1085                      | +0.1082   | 46                            | 50            | 45           | 50           |
| Prune_non_stable/100/10/2_median    | +0.0551                      | +0.0549   | 44                            | 47            | 44           | 47           |
| Prune_non_stable/100/10/2_stddev    | +1.9616                      | +1.9608   | 4                             | 12            | 4            | 12           |
| Prune_non_stable/100/10/2_cv        | +1.6718                      | +1.6716   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/10/2_pvalue    | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/10/2_mean      | +0.1152                      | +0.1149   | 167                           | 187           | 167          | 186          |
| Prune_non_stable/256/10/2_median    | +0.0467                      | +0.0466   | 167                           | 175           | 167          | 175          |
| Prune_non_stable/256/10/2_stddev    | +104.5611                    | +104.9847 | 0                             | 31            | 0            | 30           |
| Prune_non_stable/256/10/2_cv        | +93.6594                     | +94.0630  | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/10/2_pvalue   | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/10/2_mean     | +0.0213                      | +0.0212   | 2042                          | 2085          | 2037         | 2080         |
| Prune_non_stable/1024/10/2_median   | +0.0220                      | +0.0219   | 1919                          | 1961          | 1914         | 1956         |
| Prune_non_stable/1024/10/2_stddev   | +0.1037                      | +0.1036   | 344                           | 380           | 342          | 378          |
| Prune_non_stable/1024/10/2_cv       | +0.0806                      | +0.0806   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/10/2_pvalue   | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/10/2_mean     | -0.0339                      | -0.0337   | 7495                          | 7241          | 7478         | 7226         |
| Prune_non_stable/2000/10/2_median   | +0.0314                      | +0.0315   | 7021                          | 7241          | 7006         | 7226         |
| Prune_non_stable/2000/10/2_stddev   | -0.9890                      | -0.9892   | 1500                          | 16            | 1493         | 16           |
| Prune_non_stable/2000/10/2_cv       | -0.9886                      | -0.9888   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/6/4_pvalue     | 0.0140                       | 0.0140    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/6/4_mean       | +0.0515                      | +0.0513   | 52                            | 54            | 52           | 54           |
| Prune_non_stable/100/6/4_median     | +0.0544                      | +0.0542   | 48                            | 51            | 48           | 51           |
| Prune_non_stable/100/6/4_stddev     | +0.0844                      | +0.0846   | 8                             | 9             | 8            | 9            |
| Prune_non_stable/100/6/4_cv         | +0.0313                      | +0.0317   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/6/4_pvalue     | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/6/4_mean       | +0.0424                      | +0.0424   | 219                           | 228           | 219          | 228          |
| Prune_non_stable/256/6/4_median     | +0.0448                      | +0.0447   | 205                           | 215           | 205          | 214          |
| Prune_non_stable/256/6/4_stddev     | -0.1719                      | -0.1720   | 44                            | 37            | 44           | 36           |
| Prune_non_stable/256/6/4_cv         | -0.2056                      | -0.2057   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/6/4_pvalue    | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/6/4_mean      | +0.0917                      | +0.0916   | 2584                          | 2821          | 2578         | 2814         |
| Prune_non_stable/1024/6/4_median    | +0.0255                      | +0.0256   | 2585                          | 2651          | 2579         | 2645         |
| Prune_non_stable/1024/6/4_stddev    | +143.7263                    | +144.4257 | 4                             | 542           | 4            | 539          |
| Prune_non_stable/1024/6/4_cv        | +131.5702                    | +132.2224 | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/6/4_pvalue    | 0.0036                       | 0.0036    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/6/4_mean      | -0.0125                      | -0.0123   | 15285                         | 15094         | 15245        | 15057        |
| Prune_non_stable/2000/6/4_median    | +0.0099                      | +0.0099   | 14017                         | 14155         | 13984        | 14122        |
| Prune_non_stable/2000/6/4_stddev    | -0.2598                      | -0.2597   | 4055                          | 3002          | 4035         | 2987         |
| Prune_non_stable/2000/6/4_cv        | -0.2505                      | -0.2504   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/8/4_pvalue     | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/8/4_mean       | +0.0338                      | +0.0337   | 50                            | 52            | 50           | 52           |
| Prune_non_stable/100/8/4_median     | +0.0380                      | +0.0377   | 47                            | 49            | 47           | 49           |
| Prune_non_stable/100/8/4_stddev     | -0.3400                      | -0.3397   | 10                            | 7             | 10           | 7            |
| Prune_non_stable/100/8/4_cv         | -0.3616                      | -0.3612   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/8/4_pvalue     | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/8/4_mean       | +0.0011                      | +0.0012   | 225                           | 225           | 224          | 224          |
| Prune_non_stable/256/8/4_median     | +0.0189                      | +0.0189   | 210                           | 214           | 210          | 214          |
| Prune_non_stable/256/8/4_stddev     | -0.0385                      | -0.0385   | 34                            | 33            | 34           | 33           |
| Prune_non_stable/256/8/4_cv         | -0.0396                      | -0.0397   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/8/4_pvalue    | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/8/4_mean      | -0.0323                      | -0.0322   | 2924                          | 2829          | 2917         | 2823         |
| Prune_non_stable/1024/8/4_median    | +0.0057                      | +0.0057   | 2813                          | 2829          | 2807         | 2823         |
| Prune_non_stable/1024/8/4_stddev    | -0.9879                      | -0.9881   | 355                           | 4             | 353          | 4            |
| Prune_non_stable/1024/8/4_cv        | -0.9875                      | -0.9877   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/8/4_pvalue    | 0.0004                       | 0.0003    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/8/4_mean      | +0.0706                      | +0.0706   | 13803                         | 14777         | 13768        | 14740        |
| Prune_non_stable/2000/8/4_median    | +0.0051                      | +0.0053   | 13807                         | 13877         | 13772        | 13845        |
| Prune_non_stable/2000/8/4_stddev    | +57.5063                     | +58.1010  | 33                            | 1937          | 33           | 1928         |
| Prune_non_stable/2000/8/4_cv        | +53.6494                     | +54.2037  | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/10/4_pvalue    | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/10/4_mean      | +0.0352                      | +0.0350   | 55                            | 57            | 55           | 57           |
| Prune_non_stable/100/10/4_median    | +0.0405                      | +0.0402   | 51                            | 53            | 51           | 53           |
| Prune_non_stable/100/10/4_stddev    | -0.1277                      | -0.1271   | 10                            | 9             | 10           | 9            |
| Prune_non_stable/100/10/4_cv        | -0.1573                      | -0.1567   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/10/4_pvalue    | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/10/4_mean      | +0.0240                      | +0.0240   | 225                           | 231           | 225          | 230          |
| Prune_non_stable/256/10/4_median    | +0.0261                      | +0.0260   | 211                           | 217           | 211          | 216          |
| Prune_non_stable/256/10/4_stddev    | -0.1825                      | -0.1823   | 39                            | 32            | 39           | 32           |
| Prune_non_stable/256/10/4_cv        | -0.2016                      | -0.2015   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/10/4_pvalue   | 0.0046                       | 0.0046    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/10/4_mean     | +0.0048                      | +0.0048   | 2956                          | 2970          | 2949         | 2963         |
| Prune_non_stable/1024/10/4_median   | +0.0077                      | +0.0076   | 2770                          | 2792          | 2764         | 2785         |
| Prune_non_stable/1024/10/4_stddev   | -0.3259                      | -0.3255   | 585                           | 394           | 582          | 392          |
| Prune_non_stable/1024/10/4_cv       | -0.3292                      | -0.3287   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/10/4_pvalue   | 0.0312                       | 0.0211    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/10/4_mean     | -0.0633                      | -0.0631   | 14148                         | 13252         | 14111        | 13221        |
| Prune_non_stable/2000/10/4_median   | +0.0015                      | +0.0016   | 13224                         | 13243         | 13191        | 13212        |
| Prune_non_stable/2000/10/4_stddev   | -0.9919                      | -0.9918   | 2947                          | 24            | 2932         | 24           |
| Prune_non_stable/2000/10/4_cv       | -0.9913                      | -0.9912   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/6/6_pvalue     | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/6/6_mean       | +0.0957                      | +0.0955   | 59                            | 65            | 59           | 65           |
| Prune_non_stable/100/6/6_median     | +0.0284                      | +0.0283   | 59                            | 61            | 59           | 61           |
| Prune_non_stable/100/6/6_stddev     | +92.9053                     | +93.9491  | 0                             | 10            | 0            | 10           |
| Prune_non_stable/100/6/6_cv         | +84.6999                     | +85.6759  | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/6/6_pvalue     | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/6/6_mean       | +0.0284                      | +0.0281   | 283                           | 291           | 283          | 291          |
| Prune_non_stable/256/6/6_median     | +0.0348                      | +0.0345   | 265                           | 274           | 264          | 273          |
| Prune_non_stable/256/6/6_stddev     | -0.1225                      | -0.1222   | 60                            | 52            | 59           | 52           |
| Prune_non_stable/256/6/6_cv         | -0.1467                      | -0.1462   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/6/6_pvalue    | 0.0173                       | 0.0140    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/6/6_mean      | +0.0083                      | +0.0083   | 6722                          | 6778          | 6705         | 6761         |
| Prune_non_stable/1024/6/6_median    | +0.0097                      | +0.0096   | 6307                          | 6368          | 6292         | 6353         |
| Prune_non_stable/1024/6/6_stddev    | +0.0006                      | +0.0008   | 1075                          | 1076          | 1070         | 1071         |
| Prune_non_stable/1024/6/6_cv        | -0.0077                      | -0.0074   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/6/6_pvalue    | 0.0312                       | 0.0312    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/6/6_mean      | +0.0128                      | +0.0128   | 168750                        | 170916        | 168408       | 170559       |
| Prune_non_stable/2000/6/6_median    | +0.0200                      | +0.0199   | 167211                        | 170547        | 166873       | 170194       |
| Prune_non_stable/2000/6/6_stddev    | -0.8699                      | -0.8697   | 14392                         | 1872          | 14359        | 1872         |
| Prune_non_stable/2000/6/6_cv        | -0.8716                      | -0.8713   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/8/6_pvalue     | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/8/6_mean       | +0.0424                      | +0.0423   | 65                            | 68            | 65           | 68           |
| Prune_non_stable/100/8/6_median     | +0.0437                      | +0.0436   | 61                            | 64            | 61           | 63           |
| Prune_non_stable/100/8/6_stddev     | +0.4430                      | +0.4433   | 9                             | 13            | 9            | 13           |
| Prune_non_stable/100/8/6_cv         | +0.3844                      | +0.3848   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/8/6_pvalue     | 0.0028                       | 0.0028    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/8/6_mean       | +0.0322                      | +0.0319   | 285                           | 294           | 284          | 293          |
| Prune_non_stable/256/8/6_median     | +0.0362                      | +0.0358   | 267                           | 276           | 266          | 276          |
| Prune_non_stable/256/8/6_stddev     | -0.1853                      | -0.1849   | 58                            | 47            | 57           | 47           |
| Prune_non_stable/256/8/6_cv         | -0.2107                      | -0.2101   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/8/6_pvalue    | 0.0002                       | 0.0002    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/8/6_mean      | +0.0778                      | +0.0778   | 8621                          | 9291          | 8599         | 9268         |
| Prune_non_stable/1024/8/6_median    | +0.0117                      | +0.0118   | 8612                          | 8713          | 8591         | 8692         |
| Prune_non_stable/1024/8/6_stddev    | +82.0939                     | +82.4992  | 22                            | 1837          | 22           | 1828         |
| Prune_non_stable/1024/8/6_cv        | +76.0948                     | +76.4742  | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/8/6_pvalue    | 0.9698                       | 0.9698    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/8/6_mean      | -0.0353                      | -0.0351   | 227461                        | 219431        | 226950       | 218982       |
| Prune_non_stable/2000/8/6_median    | -0.0025                      | -0.0025   | 219972                        | 219422        | 219529       | 218973       |
| Prune_non_stable/2000/8/6_stddev    | -0.9664                      | -0.9663   | 52004                         | 1749          | 51741        | 1746         |
| Prune_non_stable/2000/8/6_cv        | -0.9651                      | -0.9650   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/100/10/6_pvalue    | 0.0022                       | 0.0022    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/100/10/6_mean      | +0.0329                      | +0.0330   | 63                            | 65            | 63           | 65           |
| Prune_non_stable/100/10/6_median    | +0.0338                      | +0.0338   | 59                            | 61            | 59           | 61           |
| Prune_non_stable/100/10/6_stddev    | +0.0200                      | +0.0198   | 12                            | 13            | 12           | 13           |
| Prune_non_stable/100/10/6_cv        | -0.0125                      | -0.0128   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/256/10/6_pvalue    | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/256/10/6_mean      | +0.0243                      | +0.0243   | 280                           | 287           | 279          | 286          |
| Prune_non_stable/256/10/6_median    | +0.0279                      | +0.0277   | 262                           | 269           | 261          | 268          |
| Prune_non_stable/256/10/6_stddev    | +0.2101                      | +0.2129   | 43                            | 52            | 42           | 51           |
| Prune_non_stable/256/10/6_cv        | +0.1814                      | +0.1841   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/1024/10/6_pvalue   | 0.0173                       | 0.0173    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/1024/10/6_mean     | +0.0063                      | +0.0067   | 8545                          | 8599          | 8519         | 8575         |
| Prune_non_stable/1024/10/6_median   | +0.0078                      | +0.0081   | 8010                          | 8073          | 7986         | 8051         |
| Prune_non_stable/1024/10/6_stddev   | +0.2241                      | +0.2237   | 1374                          | 1682          | 1369         | 1675         |
| Prune_non_stable/1024/10/6_cv       | +0.2164                      | +0.2156   | 0                             | 0             | 0            | 0            |
| Prune_non_stable/2000/10/6_pvalue   | 0.5205                       | 0.5205    | U Test, Repetitions: 10 vs 10 |               |              |              |
| Prune_non_stable/2000/10/6_mean     | +0.0198                      | +0.0196   | 338431                        | 345125        | 337709       | 344340       |
| Prune_non_stable/2000/10/6_median   | -0.0025                      | -0.0026   | 344612                        | 343762        | 343882       | 342989       |
| Prune_non_stable/2000/10/6_stddev   | +0.5899                      | +0.5902   | 18372                         | 29209         | 18337        | 29159        |
| Prune_non_stable/2000/10/6_cv       | +0.5590                      | +0.5596   | 0                             | 0             | 0            | 0            |
| OVERALL_GEOMEAN                     | +0.0376                      | +0.0375   | 0                             | 0             | 0            | 0            |

</details>

Reply:
The difference seems larger than I hoped :disappointed: 
Let's ignore this comment and keep the code as you have it then.

Reply:
Ok, even though I am a bit surprised by the results. I would have thought that at least the second version comes very close to the original one.